lib/charms/lego_base_k8s/v0/lego_client.py

```diff
@@ -224,7 +224,7 @@ def _push_csr_to_workload(self, csr: str) -> None:
     def _execute_lego_cmd(self) -> bool:
         """Execute lego command in workload container."""
         process = self._container.exec(
-            self._cmd, timeout=300, working_dir="/tmp", environment=self._plugin_config
+            self._cmd, timeout=300, working_dir="/tmp", environment=self._common_config | self._plugin_config
```

Comment:
This lib is owned by the [lego k8s](https://github.com/canonical/lego-base-k8s-operator) project. Let's make the change over there, bump the lib version, then pull it here.

Reply:
opened https://github.com/canonical/lego-base-k8s-operator/pull/143